lang/java/reef-runtime-yarn/src/main/java/org/apache/reef/runtime/yarn/driver/YarnContainerManager.java

```diff
@@ -535,13 +537,28 @@ private void handleNewContainer(final Container container) {
       LOG.log(Level.FINEST, "Allocated Container: memory = {0}, core number = {1}",
           new Object[] {container.getResource().getMemory(), container.getResource().getVirtualCores()});
 
+      YarnClient client = YarnClient.createYarnClient();
+      client.init(new YarnConfiguration());
+      client.start();
+
+      Set<String> nodeLabelExpressions = null;
+
+      try {
+        nodeLabelExpressions = client.getNodeToLabels().get(container.getNodeId());
+      } catch (YarnException e) {
+        e.printStackTrace();
+      } catch (IOException e) {
+        e.printStackTrace();
+      }
```

Comment:
Thank you! I will update it, except making `nodeLabelExpressions` as a `final` variable.